underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java

```diff
@@ -563,10 +563,21 @@ public InputStream open(String path, OpenOptions options) throws IOException {
     while (retryPolicy.attempt()) {
       try {
         FSDataInputStream inputStream = hdfs.open(new Path(path));
-        // pread API instead of seek is more efficient for FSDataInputStream.
-        // A seek on FSDataInputStream uses a skip op which is implemented as read + discard
-        // and hence ends up reading extra data from the datanode.
-        return new HdfsPositionedUnderFileInputStream(inputStream, options.getOffset());
+        if (options.getPositionShort()) {
+          LOG.info("AMDEBUG: using pread");
+          // pread API instead of seek is more efficient for FSDataInputStream.
+          // A seek on FSDataInputStream uses a skip op which is implemented as read + discard
+          // and hence ends up reading extra data from the datanode.
+          return new HdfsPositionedUnderFileInputStream(inputStream, options.getOffset());
```

Comment:
Do we end up caching this stream anywhere, if so would it be a problem if it is a certain type?

Reply:
I just grepped our codebase, doesn't look like we do, so we should be fine